### DISCUSSION TITLE
Shift+Tab should go delete a level of indentation in Insert mode

### DISCUSSION THREAD

**Discussion by jroth5** [helix-editor/helix#6972] [Ideas]
Shift+Tab is somewhat common keyboard shortcut that essentially removes a level of indentation. While many popular applications support this shortcut, implementations vary. 

In VSCode, Shift+Tab decreases the level of indentation of the entire line, regardless of where the cursor is. In Visual Studio, it deletes a tab's-worth of whitespace behind the cursor (and does nothing if there isn't whitespace behind the cursor). Some applications do nothing, some just treat it as a normal tab. 

I suggest we follow the behavior of Visual Studio and just delete a tab of whitespace behind the cursor. If the user wants to decrease the line's indentation, they should use < in Normal mode in my opinion.

Currently, Shift+Tab does nothing in Helix.

P.S. If you're curious why Shift+Tab would behave like this, you can think of the Tab key as telling the rest of the keyboard to do the other symbol on keys pressed. For example, the 2 key instead makes the @ symbol while Shift is held since that key has a "2" and a "@" on it. On many keyboards, the Tab key has a right arrow and a left arrow. If normal tab goes right, holding shift should go left.

**Comment by CptPotato** [comment]
You can do this by adding a key binding to your config:

```toml
[keys.insert]
S-tab = "unindent"
```

**Reply by jroth5** [reply]
That works for the VSCode-style of Shift+Tab, but there doesn't seem to be a way to delete a tab of whitespace behind the cursor. I created a "remove_tab" command on my own branch that does this. Would it be worth creating a PR for that command if it wouldn't be used by a default?

Also, what goes into deciding whether this keybind would be worth making a default? Not directly advocating for this to be a default -- mostly curious about the project's design process in general.

**Reply by pdppdp** [reply]
This is a much used feature in other programming editors, such as PFE, Notepad++, VisualStudio.   Especially useful in structured comment blocks with the now-pervasive "\\\\" comment syntax.  Even if it lacks a default.  It's functional lack is problem, even if some do not perceive.

For the world to move forward, it needs to actually move forward, not backward.

**Reply by markstos** [reply]
Note that `unident` already have a normal mode binding of "<".
